src/object-store/src/layers/prometheus.rs

```diff
@@ -410,15 +438,35 @@ pub struct PrometheusMetricWrapper<R> {
     inner: R,
 
     op: Operation,
-    scheme: String,
+    bytes_counter: Histogram,
+    requests_duration_seconds: Histogram,
+
+    start: Instant,
+    bytes: u64,
+}
+
+impl<R> Drop for PrometheusMetricWrapper<R> {
+    fn drop(&mut self) {
+        self.bytes_counter.observe(self.bytes as f64);
+        let dur = self.start.elapsed().as_secs_f64();
+        self.requests_duration_seconds.observe(dur);
+    }
 }
 
 impl<R> PrometheusMetricWrapper<R> {
-    fn new(inner: R, op: Operation, scheme: &String) -> Self {
+    fn new(
+        inner: R,
+        op: Operation,
+        bytes_counter: Histogram,
+        requests_duration_seconds: Histogram,
+    ) -> Self {
```

Comment:
I considered passing `HistogramTimer` rather than using `Instant`, but `opendal` use `Instant` style and I think it's OK as `bytes_counter` is also a `Histogram` so I follow this style.